components/table/ReportTable.tsx

```diff
@@ -102,6 +111,57 @@ const entriesByTask = (entries: TimesheetEntry[], taskId: number) =>
 const taskByProject = (tasks: Task[], projectId: number) =>
     tasks.filter((task) => task.project.id === projectId)
 
+function DateInput({ setStartDate, setEndDate }: DateInputProps): JSX.Element {
+    const dateRegex = /^\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])$/
+    const validateDate = (date: string): boolean => dateRegex.test(date)
+
+    const [newStartDate, setNewStartDate] = useState<string>("")
+    const handleStartDateChange = (event: React.FormEvent<HTMLInputElement>) =>
+        setNewStartDate(event.currentTarget.value)
+
+    const [newEndDate, setNewEndDate] = useState<string>("")
+    const handleEndDateChange = (event: React.FormEvent<HTMLInputElement>) =>
+        setNewEndDate(event.currentTarget.value)
+
+    const handleOnClick = () => {
+        if (
+            validateDate(newStartDate) &&
+            validateDate(newEndDate) &&
+            newStartDate < newEndDate
+        ) {
+            setStartDate(newStartDate)
+            setEndDate(newEndDate)
+        } else {
+            throw Error("Invalid dates")
+        }
+    }
+
+    return (
+        <>
+            <InputGroup>
+                <Input
+                    placeholder="Start date (YYYY-MM-DD)"
+                    value={newStartDate}
+                    onChange={handleStartDateChange}
+                ></Input>
+                <Input
+                    placeholder="End date (YYYY-MM-DD)"
+                    value={newEndDate}
+                    onChange={handleEndDateChange}
+                ></Input>
```

Comment:
The inputs should be of type date so you don't need all the regex validation for the date being valid. 
See the form for creating a new project for example https://github.com/three-consulting/epoc-frontend/blob/main/components/form/ProjectForm.tsx#L147-L179
Can be found at url `/project/new`

Reply:
@ToukoH you could still validate that the start date is before the end date, when both are chosen. You can use the `FormErrorMessage` component to display the error message: https://chakra-ui.com/docs/components/form/form-control#error-message